src/core.ts

```diff
@@ -197,25 +185,19 @@ export function onWorkerMessage(evt: MessageEvent) {
     switch (message.type) {
       case WorkerMessageType.CompressedBatch:
         let uploadMsg = message as ICompressedBatchMessage;
-        let onSuccess = (status: number) => { mapProperties(droppedPayloads, uploadDroppedPayloadsMappingFunction, true); };
-        let onFailure = (status: number) => { onFirstSendDeliveryFailure(status, uploadMsg.rawData, uploadMsg.compressedData); };
         if (queueUploads) {
-          let uploadInfo: IUploadInfo = {
-            payload: uploadMsg.compressedData,
-            onSuccess,
-            onFailure
-          };
-          pendingUploads.push(uploadInfo);
+          enqueueUpload(uploadMsg.compressedData, uploadMsg.rawData);
         } else {
-          upload(uploadMsg.compressedData, onSuccess, onFailure);
+          upload(uploadMsg.compressedData, uploadMsg.rawData);
         }
+        sequence = uploadMsg.rawData.envelope.sequenceNumber + 1;
```

Comment:
>sequence = uploadMsg.rawData.envelope.sequenceNumber + 1; [](start = 8, length = 57)

what is sequence used for? I only saw it being initialized and set here, never consumed right?

Reply:
It keeps track of what is the sequence number of the next batch, so that when teardown happens and compression worker shuts down, core can upload remaining events itself, wrapped in an envelope with a correct sequence number. It is used in the "uploadPendingEvents" function on line 233.

---
In reply to: [167967974](https://github.com/Microsoft/clarity-js/pull/63#discussion_r167967974) [](ancestors = 167967974)